src/sentry/web/api.py

```diff
@@ -94,35 +101,54 @@ def wrapped(request, *args, **kwargs):
     return wrapped
 
 
+def count_stats(org_id, project_id, key_id, outcome, reason=None):
+    # First increment all the legacy TSDB counters.
+    increment_list = [
+        (tsdb.models.project_total_received, project_id),
+        (tsdb.models.organization_total_received, org_id),
+        (tsdb.models.key_total_received, key_id),
+    ]
+    if outcome == 'filtered':
+        increment_list.extend([
+            (tsdb.models.project_total_blacklisted, project_id),
+            (tsdb.models.organization_total_blacklisted, org_id),
+            (tsdb.models.key_total_blacklisted, key_id),
+        ])
+        if reason in FILTER_STAT_KEYS_TO_VALUES:
+            increment_list.append((FILTER_STAT_KEYS_TO_VALUES[reason], project_id))
+
+    elif outcome == 'rate_limited':
```

Comment:
unclear if outcome=filtered and outcome=rate_limited should both just be outcome=rejected (ie outcome is boolean) with a different `reason` for each

Reply:
that way outcome would be binary

Reply:
Product-wise, I think it makes sense to split them. All of our graphs are going to have to roll up based on these outcomes, and the hovercard details would split by reason, so it's easier for me to see queries that group by outcome and reason, and the UI possibly flattening them.

Reply:
yeah I think adding outcome=invalid also led me to this conclusion.